metabase/metabase
# Move Google Auth toggle to the auth card

Epic https://github.com/metabase/metabase/issues/19428

Changes:
- Adds a toggle to enable/disable google auth independently to the settings themselves
- Makes `Client ID` required
- When not enabled, there are 2 buttons in the form `Save and enable` and `Save but don't enable`
- Uses `google-auth-enabled` instead of checking for `google-auth-client-id` in other places

How to test:
- Go to Admin -> Settings -> Authentication -> Google
- Enter `example.apps.googleusercontent.com` as Client ID and click `Save and enable`
- Go to Admin -> Settings -> Authentication and make sure the toggle is visible and enabled
- Make sure it is possible to toggle it on and off

<img width="539" alt="Screenshot 2022-09-26 at 19 47 39" src="https://user-images.githubusercontent.com/8542534/192335251-d444469d-9aa7-4f16-bccf-bab8b139d9dd.png">
<img width="604" alt="Screenshot 2022-09-26 at 19 47 46" src="https://user-images.githubusercontent.com/8542534/192335261-43248cd6-fd0e-4290-a284-5b0f22bb7fef.png">
